packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -1311,8 +1332,19 @@ class _CupertinoActionSheetActionState extends State<CupertinoActionSheetAction>
     final double verticalPadding = _kActionSheetButtonVerticalPaddingBase
         + fontSize * _kActionSheetButtonVerticalPaddingFactor;
 
+    final Set<WidgetState> states = <WidgetState>{
```

Comment:
Detecting when a `WidgetState` is active is our responsibility. Typically it's done by having different flags that are activated depending on some metric. For example, if we want to track the `scrolledUnder` state, we would have to get the ScrollMetrics of the scroll view and do the math to calculate when the widget has been scrolled under (and when it has, add `WidgetState.scrolledUnder` to `states`.

It's more complex logic than anticipated, so it's okay to leave it out of this PR and handle it subsequently on a case-by-case basis (for each relevant state).

Reply:
oh ok, so do you want me to rollback to the original pr?

Reply:
marking as resolved taken care in recent commit